packages/pluggableWidgets/progress-circle-web/src/ProgressCircle.editorConfig.ts

```diff
@@ -0,0 +1,98 @@
+import { hidePropertiesIn, hidePropertyIn, Problem, Properties } from "@widgets-resources/piw-utils";
+import { ProgressCirclePreviewProps, TypeEnum } from "../typings/ProgressCircleProps";
+
+type PreviewPropsKey = keyof ProgressCirclePreviewProps;
+const propKeys: { [Type in TypeEnum]: PreviewPropsKey[] } = {
+    static: ["staticCurrentValue", "staticMaxValue", "staticMinValue"],
+    dynamic: ["dynamicCurrentValue", "dynamicMaxValue", "dynamicMinValue"],
+    expression: ["expressionCurrentValue", "expressionMaxValue", "expressionMinValue"]
+};
+
+export function getProperties(values: ProgressCirclePreviewProps, defaultProperties: Properties): Properties {
+    switch (values.type) {
+        case "dynamic":
+            hidePropertiesIn(defaultProperties, values, [...propKeys.static, ...propKeys.expression]);
+            break;
+        case "static":
+            hidePropertiesIn(defaultProperties, values, [...propKeys.dynamic, ...propKeys.expression]);
+            break;
+        case "expression":
+            hidePropertiesIn(defaultProperties, values, [...propKeys.static, ...propKeys.dynamic]);
+            break;
+        default:
+            break;
+    }
+
+    if (values.showLabel) {
+        if (values.labelType !== "custom") {
+            hidePropertyIn(defaultProperties, values, "customLabel");
+        }
+
+        if (values.labelType !== "text") {
+            hidePropertyIn(defaultProperties, values, "labelText");
+        }
+    } else {
+        hidePropertiesIn(defaultProperties, values, ["customLabel", "labelText", "labelType"]);
+    }
+
+    return defaultProperties;
+}
+
+function checkValue(key: PreviewPropsKey, values: ProgressCirclePreviewProps): Problem | null {
+    if (!values[key]) {
+        let message;
+        if (key.endsWith("CurrentValue")) {
+            message = "The current value property is required";
+        } else if (key.endsWith("MaxValue")) {
+            message = "The maximum value property is required";
+        } else if (key.endsWith("MinValue")) {
+            message = "The minimum value property is required";
+        }
+
+        if (message) {
+            return {
+                property: key,
+                message
+            };
+        }
+    }
+    return null;
+}
+
+export function check(values: ProgressCirclePreviewProps): Problem[] {
+    const errors: Problem[] = [];
+
+    if (values.type === "dynamic" || values.type === "expression") {
+        propKeys[values.type].forEach(key => {
+            const error = checkValue(key, values);
+            if (error) {
+                errors.push(error);
+            }
+        });
+    }
+
+    if (values.type === "static") {
+        if (
+            values.staticCurrentValue !== null &&
+            values.staticMinValue !== null &&
+            values.staticCurrentValue < values.staticMinValue
```

Comment:
The message below says that the current value can be greater than or equal to min value; should this operator be `<=`?